src/models/plot.ts

```diff
@@ -108,10 +134,66 @@ export type TraceConfig = {
   legendwidth?: number;
   url_pattern?: string;
   //queryPattern will be deprecated 
-  queryPattern?:string,
+  queryPattern?:string;
   response_format?: 'csv' & 'json';
 };
 
+/**
+ * Breakpoint config
+ */
+export type BreakpointConfig =
+{
+  lg: number; 
+  md: number; 
+  sm: number; 
+  xs: number;
+}
+
+/**
+ * Margin/Padding config
+ */
+export type MarginPaddingConfig =
+{
+  lg: [number, number]; 
+  md: [number, number]; 
+  sm: [number, number]; 
+  xs: [number, number];
+}
+
+/**
+ * Grid Layout config
+ */
+export type GridLayoutConfig = {
```

Comment:
Does `ReactGridLayout` have types available in the library we are including?

Reply:
Yes it does. I can replace that